client/omnichannel/directory/ContactNewEdit.js

```diff
@@ -0,0 +1,209 @@
+import React, { useState, useMemo } from 'react';
+import { Field, TextInput, Icon, ButtonGroup, Button, Box } from '@rocket.chat/fuselage';
+import { useMutableCallback } from '@rocket.chat/fuselage-hooks';
+
+
+import { useTranslation } from '../../contexts/TranslationContext';
+import VerticalBar from '../../components/VerticalBar';
+import { useForm } from '../../hooks/useForm';
+import { isEmail } from '../../../app/utils';
+import { useComponentDidUpdate } from '../../hooks/useComponentDidUpdate';
+import { useEndpointAction } from '../../hooks/useEndpointAction';
+import { useToastMessageDispatch } from '../../contexts/ToastMessagesContext';
+import { useEndpointData } from '../../hooks/useEndpointData';
+import { FormSkeleton } from './Skeleton';
+import CustomFieldsForm from '../../components/CustomFieldsForm';
+import { hasAtLeastOnePermission } from '../../../app/authorization';
+import { UserAutoComplete } from '../../components/AutoComplete';
+import { AsyncStatePhase } from '../../hooks/useAsyncState';
+
+
+const initialValues = {
+	name: '',
+	email: '',
+	phone: '',
+	username: '',
+};
+
+const getInitialValues = (data) => {
+	if (!data) {
+		return initialValues;
+	}
+
+	const { contact: { name, phone, visitorEmails, livechatData, user } } = data;
+
+	return {
+		name: name ?? '',
+		email: visitorEmails ? visitorEmails[0].address : '',
+		phone: phone ? phone[0].phoneNumber : '',
+		livechatData: livechatData ?? '',
+		username: user?.username ?? '',
+	};
+};
+
+export function ContactEditWithData({ id, reload, close }) {
```

Comment:
Why not using the same method/function `ContactNewEdit` ?

Reply:
It's using the same approach of `AgentEdit.js`, `DepartmentEdit.js`, etc, because the endpoint callers are using hooks, so you cannot do a condition to call a HTTP Request

Reply:
I got it, thanks.